src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -958,7 +958,53 @@ class AlterTableTableColumnActionSegment(BaseSegment):
 
     match_grammar = Sequence(
         OneOf(
-            # @TODO: Add Column
+            # Add Column
+            Sequence(
+                "ADD",
+                "COLUMN",
+                Ref("ColumnReferenceSegment"),
+                Ref("DatatypeSegment"),
+                OneOf(
+                    # Default
+                    Sequence(
+                        "DEFAULT",
+                        Ref("ExpressionSegment"),
+                    ),
+                    # Auto-increment/identity column
+                    Sequence(
+                        OneOf(
+                            Ref.keyword("AUTOINCREMENT"),
+                            Ref.keyword("IDENTITY"),
+                        ),
+                        OneOf(
+                            # ( <start_num>, <step_num> )
+                            Bracketed(
+                                Ref("NumericLiteralSegment"),
+                                Ref("CommaSegment"),
+                                Ref("NumericLiteralSegment"),
+                            ),
+                            # START <num> INCREMENT <num>
+                            Sequence(
+                                "START",
+                                Ref("NumericLiteralSegment"),
+                                "INCREMENT",
+                                Ref("NumericLiteralSegment"),
+                            ),
+                            optional=True,
+                        ),
+                    ),
+                    optional=True,
+                ),
+                # @TODO: Add support for `inlineConstraint`
+                Sequence(
+                    Ref.keyword("WITH", optional=True),
+                    "MASKING",
+                    "POLICY",
+                    Ref("ObjectReferenceSegment"),
+                    # @TODO: Add support for delimited col/expression list
+                    optional=True,
```

Comment:
As this is part of `OneOf` do you need the `optional=True,`?

```suggestion
```

Reply:
I _think_ we need this since it's not part of the first `OneOf` (Default / Autoincrement), and it makes `WITH MASKING POLICY` clause optional.

* Screenshot of `ADD COLUMN` structure:
![image](https://user-images.githubusercontent.com/79668578/149795434-cab70c20-c88f-4b70-95d9-d9e1f07874c1.png)
* Referenced grammar: * Referenced grammar: https://docs.snowflake.com/en/sql-reference/sql/alter-table.html

Reply:
Ah sorry it's part of a `Sequence` not part of a `OneOf`. OK resolving.